test/groovy/PiperExecuteBinTest.groovy

```diff
@@ -154,6 +154,31 @@ class PiperExecuteBinTest extends BasePiperTest {
         assertThat(artifacts[0], allOf(hasEntry('artifacts', '1234.pdf'), hasEntry('allowEmptyArchive', false)))
     }
 
+    @Test
+    void testPiperExecuteBinANSCredentials() {
+        shellCallRule.setReturnValue('./piper getConfig --contextConfig --stepMetadata \'.pipeline/tmp/metadata/test.yaml\'', '{"fileCredentialsId":"credFile", "tokenCredentialsId":"credToken", "credentialsId":"credUsernamePassword", "dockerImage":"my.Registry/my/image:latest"}')
+
+        def newScript = nullScript
+        newScript.commonPipelineEnvironment.configuration.hooks = [ans: [serviceKeyCredentialsId: "ansServiceKeyID"]]
+
+        List stepCredentials = [
+        ]
+        stepRule.step.piperExecuteBin(
```

Comment:
I'd expect to  test the handleANSCredentials method here:
1. list nil
2. list empty
3. list with content

Reply:
1. The list given to the handleANSCredentials method comes from the step call itself. If it is nil, it breaks everything (I don't think I need a test for that since I think it's always given to the step, even if it's an empty list.

2. This test already covers the list being empty: https://github.com/SAP/jenkins-library/blob/da306ad68a023ef5bf809b413986993d3c086e11/test/groovy/PiperExecuteBinTest.groovy#L164

3. I added a test here for the list with content: https://github.com/SAP/jenkins-library/blob/da306ad68a023ef5bf809b413986993d3c086e11/test/groovy/PiperExecuteBinTest.groovy#L182

4. I added a test for the serviceKeyCredentialsId being null here:  https://github.com/SAP/jenkins-library/blob/da306ad68a023ef5bf809b413986993d3c086e11/test/groovy/PiperExecuteBinTest.groovy#L236

I added a test for the serviceKeyCredentialsId being empty here: https://github.com/SAP/jenkins-library/blob/da306ad68a023ef5bf809b413986993d3c086e11/test/groovy/PiperExecuteBinTest.groovy#L213

I added a test for the hooks section being empty (no ans section) here: https://github.com/SAP/jenkins-library/blob/da306ad68a023ef5bf809b413986993d3c086e11/test/groovy/PiperExecuteBinTest.groovy#L259

This test already covers the serviceKeyCredentialsId having content: https://github.com/SAP/jenkins-library/blob/da306ad68a023ef5bf809b413986993d3c086e11/test/groovy/PiperExecuteBinTest.groovy#L158